core/stream_group.go

```diff
@@ -70,7 +70,8 @@ func (g *StreamGroup) handleRoute(hf *frame.HandshakeFrame, md metadata.M) (rout
 			return nil, err
 		}
 		if ok {
-			stream.Close()
+			// if same name sfn here, close the old connection.
+			stream.Conn().CloseWithError(e.Error())
```

Comment:
If it will be better if send a `GoawayFrame` here instead of just closing the stream with an error, as this stream(the old one) will be removed from the router.

By this, also, `core/client.go` do not need `strings.HasSuffix(err.Error(), "is already linked to another stream")` to determine if should `reconnect` to zipper.

There are two things zipper should do:

1. Tell the client that you should close the connection, and DO NOT reconnect.
2. Close the stream in case of client didn't do it, reclaim computing resources promptly.